components/i2c/include/i2c.hpp

```diff
@@ -303,6 +306,7 @@ class I2c : public espp::BaseComponent {
   Config config_;
   bool initialized_ = false;
   std::mutex mutex_;
+  std::condition_variable init_done_cv_; ///< Signal for when I2C init is done (when pinned to 2nd core)
```

Comment:
Since this is only used in init, it could exist within the scope of that function, without needing to be in the class and persist for the life of the class.